### ISSUE TITLE
Auto Updater with no ability to disable remote downloads

### ISSUE THREAD

**Issue by chrismwiggs** [GeyserMC/Geyser#2530]
### Describe the bug

From what I can tell, and I am fully ready to accept this as a misunderstanding, it appears Geyser auto-updates with no option to disable the functionality. 

I noticed this because I run a version of geyser with modified mappings (for a specific issue, which I have another issue open for), and my modified mappings routinely get overwritten. At first, I thought I was going crazy, but after carefully reading the console, I see some sort of auto-update process. Maybe I am misunderstanding and there is just some other issue causing my mappings to vanish, but this makes a lot of sense having seen this log. 

### To Reproduce

Run geyser, wait for an update to be released.

### Expected behaviour

Plugins absolutely, under no circumstances, should download and run code autonomously from remote sources. Full stop. 

This needs to have a config option to disable it. I understand why you would do this (the majority of users probably WANT these auto-updates, and rely on them to keep things running smoothly) and why you might want a config option set to true by default, but I just feel like software (specifically plugins) should never be allowed to make arbitrary changes to its own code with no action or confirmation by a user/admin. Bigger projects have had functionality like this flagged as malicious and a potential security risk.

### Geyser Version

1.4.3-SNAPSHOT

### Additional Context

I'm hoping I am just misunderstanding what is going on here; that means there is some other mysterious issue that is causing my mappings to reset (which would be worth looking into, but far less important). I don't mean to come off so strong, but I think it's super important to be able to trust the plugins you're relying on, and in my opinion, that means having full control over what is coming in and out of your machine.

**Comment by Jens-Co** [comment]
Hello there, Currently there is no auto updating feature within geyser. If your build gets overwritten/ updated make sure you are not running an plugin that is able to update geyser like ( GeyserUpdater ) or a skript. Can you please upload a dump to? ( console command : geyser dump )

**Comment by jackson-57** [comment]
AFAIK the only resource Geyser downloads from the internet is the Minecraft Java locale file.

**Comment by chrismwiggs** [comment]
Here is a dump: https://dump.geysermc.org/mmMy3E8GZOiWK89TsWIB7OmbAf8sYElz

Maybe what I was seeing was something related to updating to Java 16? It was just from restarting the geyser instance with an old (pre java 16) build, referencing what looked like a git commit.

**Comment by Chew** [comment]
Can you provide what the actual console log said (even just a snippet, but not a paraphrase)?

Unless the project changed under my nose, we have not implemented an auto-updater. See #1333 and #1335 and the [FAQ](https://github.com/GeyserMC/Geyser/wiki/FAQ#how-can-i-auto-update-geyser).

Based on the wording found [here](https://github.com/GeyserMC/Geyser/wiki/FAQ#can-i-use-geyser-with-pterodactyl-panel), the pterodacytl egg (if you are using it) _might_ be auto-updating, but it's hard to tell.

**Comment by chrismwiggs** [comment]
I think you might be on to something with the egg updating. I would usually include the log for something like this, but can't find any reference to it in any of the Geyser logs. Which leads me to believe that what I saw was just the egg, because that wouldn't be captured in the Geyser logs. 

Still confused why my mappings are resetting, but that might be an issue for another day. (t=The whole reason this started was because players were reporting that my updated particle mappings seemed to have been reverted to the Geyser defaults, despite me not touching it for ~2 weeks.)
